flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/PushProjectIntoTableSourceScanRule.java

```diff
@@ -139,10 +139,23 @@ public void onMatch(RelOptRuleCall call) {
         final FlinkTypeFactory typeFactory = unwrapTypeFactory(scan);
         final ResolvedSchema schema = sourceTable.contextResolvedTable().getResolvedSchema();
         final RowType producedType = createProducedType(schema, sourceTable.tableSource());
-        final NestedSchema projectedSchema =
+        NestedSchema projectedSchema =
                 NestedProjectionUtil.build(
                         getProjections(project, scan),
                         typeFactory.buildRelNodeRowType(producedType));
+        // we can not perform an empty column query to the table scan, just choose the first column
+        // in such case
+        if (projectedSchema.columns().isEmpty()) {
+            if (scan.getRowType().getFieldCount() == 0) {
+                return;
```

Comment:
is this case valid ?

Reply:
I was going to add an assertion here, just a protection for the following lines